irqbalance.c

```diff
@@ -360,14 +366,16 @@ gboolean sock_handle(gint fd, GIOCondition condition, gpointer user_data __attri
 	int recv_size = 0;
 	int valid_user = 0;
 
+	struct cmsghdr *cmsg;
 	struct iovec iov = { buff, 500 };
 	struct msghdr msg = { 0 };
 	msg.msg_iov = &iov;
 	msg.msg_iovlen = 1;
 	msg.msg_control = malloc(CMSG_SPACE(sizeof(struct ucred)));
 	msg.msg_controllen = CMSG_SPACE(sizeof(struct ucred));
 
-	struct cmsghdr *cmsg;
+	if (!msg.msg_control)
+		return FALSE;
```

Comment:
I don't think this is right, as it will cancel the watch on the socket.  You still want to accept the connection and receive the message, dropping it before you return.

Reply:
ok，I see. 
I have modify the patch by checking the cmsg instead of msg.msg_control. thanks.